.changeset/soft-lobsters-promise.md

```diff
@@ -0,0 +1,5 @@
+---
+'@vanilla-extract/vite-plugin': patch
+---
+
+Fix HMR and support React Server Components.
```

Comment:
Could you please split this into two changesets?

1) `patch` release: `Fix HMR`
2) `minor` release: `Add [experimental support for React Server Components](https://github.com/facebook/react/pull/22952)`

Reply:
Done 👍 
If you want I can split the PRs as well (extract the env variable in a new PR).
Originally, I did it in one PR because at first I thought the HMR was only broken for RSC 😅

Reply:
As long as it's two changesets, I'm happy 😊